common/trinary/trit_tryte_sse42.h

```diff
@@ -0,0 +1,281 @@
+#define INDEX_3DIFF_1D 0x01, 0x04, 0x07, 0x0A, 0x0D
+#define INDEX_3DIFF_2E 0x02, 0x05, 0x08, 0x0B, 0x0E
+#define REPEAT0(str)
+#define REPEAT1(str) str
+#define REPEAT2(str) REPEAT1(str), str
+#define REPEAT3(str) REPEAT2(str), str
+#define REPEAT4(str) REPEAT3(str), str
+#define REPEAT5(str) REPEAT4(str), str
+#define REPEAT6(str) REPEAT5(str), str
+#define REPEAT7(str) REPEAT6(str), str
+#define REPEAT8(str) REPEAT7(str), str
+#define REPEAT9(str) REPEAT8(str), str
+#define REPEAT10(str) REPEAT9(str), str
+#define REPEAT11(str) REPEAT10(str), str
+#define REPEAT(n, str) REPEAT##n(str)
+
+static inline void trits_to_trytes_sse42(trit_t const *const trits, tryte_t *const trytes, size_t const length) {
+  const int block_8bit = BLOCK_8BIT(__m128i);
+  const int8_t setMSB = 0x80;
+  const __m128i tryteAlphabet[2] = {
+      _mm_setr_epi8(TRYTE_ALPHABET[0], TRYTE_ALPHABET[1], TRYTE_ALPHABET[2], TRYTE_ALPHABET[3], TRYTE_ALPHABET[4],
+                    TRYTE_ALPHABET[5], TRYTE_ALPHABET[6], TRYTE_ALPHABET[7], TRYTE_ALPHABET[8], TRYTE_ALPHABET[9],
+                    TRYTE_ALPHABET[10], TRYTE_ALPHABET[11], TRYTE_ALPHABET[12], TRYTE_ALPHABET[13], TRYTE_ALPHABET[14],
+                    TRYTE_ALPHABET[15]),
+      _mm_setr_epi8(TRYTE_ALPHABET[16], TRYTE_ALPHABET[17], TRYTE_ALPHABET[18], TRYTE_ALPHABET[19], TRYTE_ALPHABET[20],
+                    TRYTE_ALPHABET[21], TRYTE_ALPHABET[22], TRYTE_ALPHABET[23], TRYTE_ALPHABET[24], TRYTE_ALPHABET[25],
+                    TRYTE_ALPHABET[26], 0, 0, 0, 0, 0)};
+  /* For shuffling the bytes of the input trits */
+  const __m128i shuffleLow[3] = {_mm_setr_epi8(REPEAT(0, setMSB) COMMA(0) INDEX_3DIFF_0F COMMA(1) REPEAT(10, setMSB)),
+                                 _mm_setr_epi8(REPEAT(6, setMSB) COMMA(1) INDEX_3DIFF_2E COMMA(1) REPEAT(5, setMSB)),
+                                 _mm_setr_epi8(REPEAT(11, setMSB) COMMA(1) INDEX_3DIFF_1D COMMA(0) REPEAT(0, setMSB))};
+  const __m128i shuffleMid[3] = {_mm_setr_epi8(REPEAT(0, setMSB) COMMA(0) INDEX_3DIFF_1D COMMA(1) REPEAT(11, setMSB)),
+                                 _mm_setr_epi8(REPEAT(5, setMSB) COMMA(1) INDEX_3DIFF_0F COMMA(1) REPEAT(5, setMSB)),
+                                 _mm_setr_epi8(REPEAT(11, setMSB) COMMA(1) INDEX_3DIFF_2E COMMA(0) REPEAT(0, setMSB))};
+  const __m128i shuffleHigh[3] = {_mm_setr_epi8(REPEAT(0, setMSB) COMMA(0) INDEX_3DIFF_2E COMMA(1) REPEAT(11, setMSB)),
+                                  _mm_setr_epi8(REPEAT(5, setMSB) COMMA(1) INDEX_3DIFF_1D COMMA(1) REPEAT(6, setMSB)),
+                                  _mm_setr_epi8(REPEAT(10, setMSB) COMMA(1) INDEX_3DIFF_0F COMMA(0) REPEAT(0, setMSB))};
+  /* The mask with interleaved bytes of 0xFF and 0x00 */
+  const __m128i byteInterMask = _mm_set_epi32(0xFF00FF00, 0xFF00FF00, 0xFF00FF00, 0xFF00FF00);
+
+  /* Start converting */
+  for (size_t i = 0; i < length / 3 / block_8bit; i++) {
+    /* Get trit data */
+    __m128i dataFirst = _mm_loadu_si128((__m128i *)(trits) + i * 3);
```

Comment:
Increment `i` by 3 in the loop to avoid `i * 3`.

Reply:
However, I would use `i` at line `125`, which does not multiply with `3`.

Reply:
I marked it as resolved since I still need to use `i` instead of `i * 3`.